### PR TITLE
Update hunting queries

### PR DESCRIPTION
**Pull request by rahul0216**
Change(s):
   - Updated parser to correctly parse failed login events
   - Added Entity mapping in all the hunting queries
   - Added version property in all the hunting queries

   Reason for Change(s):
   - Verifying hunting queries

   Version Updated:
   - Yes

   Testing Completed:
   - KQL functional testing

### CODE REVIEW

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserDeletedFromDatabase.yaml [hunk 1]
```diff
@@ -27,4 +26,25 @@ query: |
         | where Statement has "Alter role" and Statement has "drop member"
         | parse Statement with * "DROP MEMBER [" TargetUser:string "]" *
         | project TimeGenerated, Computer, Action, ClientIP, CurrentUser, DatabaseName, TargetUser, ObjectName, Statement
-        | extend timestamp = TimeGenerated, AccountCustomEntity = CurrentUser, IPCustomEntity = ClientIP  
\ No newline at end of file
+        | extend Name = iff(CurrentUser contains '@', tostring(split(CurrentUser, '@', 0)[0]), CurrentUser)
+        | extend UPNSuffix = iff(CurrentUser contains '@', tostring(split(CurrentUser, '@', 1)[0]), '')
+        | extend Name = iff(CurrentUser contains '\\', tostring(split(CurrentUser, '\\', 1)[0]), Name)
+        | extend NTDomain = iff(CurrentUser contains '\\', tostring(split(CurrentUser, '\\', 0)[0]), '')
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserDeletedFromDatabase.yaml:32]: If login through AAD is enabled, UPN will come. So I think we should handle both scenarios.

File: Parsers/SQLSever/SQLServer_Parser.txt [hunk 2]
```diff
@@ -36,10 +36,9 @@ let FailedLogon = SQlData
 | where EventLevelName has "error"
 | where RenderedDescription startswith "Login"
 | parse kind=regex RenderedDescription with "Login" LogonResult:string
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Parsers/SQLSever/SQLServer_Parser.txt:38]: I verified once again, but I don't see any extra space.

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-Failed SQL Logons.yaml [hunk 3]
```diff
@@ -13,11 +13,26 @@ tactics:
...
 query: |
-
       // SQLEvent is not the table name, it is the function name that should already be imported into your workspace.
       // The underlying table where the data exists is the Event table.
       SQLEvent
       | where LogonResult has "failed"
-      | summarize count() by TimeGenerated, CurrentUser, Reason, ClientIP
-      | extend timestamp = TimeGenerated, AccountCustomEntity = CurrentUser, IPCustomEntity = ClientIP
+      | summarize by TimeGenerated, CurrentUser, Reason, ClientIP
+      | extend Name = tostring(split(CurrentUser, '@', 0)[0]), UPNSuffix = tostring(split(CurrentUser, '@', 1)[0])
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-Failed SQL Logons.yaml]: Updated to handle AD domain login name and UPN both.

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-MultipleFailedLogon_FromSameIP.yaml [hunk 4]
```diff
@@ -25,4 +24,19 @@ query: |
         | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), TotalFailedLogons = count() by ClientIP, CurrentUser, Computer
         | where TotalFailedLogons >= failedThreshold
         | project StartTime, ClientIP, TotalFailedLogons, CurrentUser, Computer
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-MultipleFailedLogon_FromSameIP.yaml]: done

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-MultipleFailedLogon_InShortSpan.yaml [hunk 5]
```diff
@@ -25,4 +24,19 @@ query: |
         | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), TotalFailedLogons = count() by CurrentUser, ClientIP
         | where TotalFailedLogons >= failedThreshold
         | project StartTime, CurrentUser, TotalFailedLogons, ClientIP
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-MultipleFailedLogon_InShortSpan.yaml]: done.

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserRemovedFromSecurityAdmin.yaml [hunk 6]
```diff
@@ -27,4 +26,19 @@ query: |
         | parse Statement with * "DROP MEMBER [" TargetUser:string "]" *
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserRemovedFromSecurityAdmin.yaml:26]: done.

File: Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserRemovedFromServerRole.yaml [hunk 7]
```diff
@@ -18,12 +18,26 @@ relevantTechniques:
   - T1078
   - T1496
 query: |
-
         // SQLEvent is not the table name, it is the function name that should already be imported into your workspace.
         // The underlying table where the data exists is the Event table.
         // This query checks for user removed from a ServerRole
         SQLEvent
         | where Statement has "Alter Server role" and Statement has "drop member"
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft Windows SQL Server Database Audit/Hunting Queries/SQL-UserRemovedFromServerRole.yaml]: done

**Review by vakohl — APPROVED** [review]

**Comment by vakohl** [comment]
@v-atulyadav FYI. This is approved.

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**